aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -148,7 +153,14 @@ module aptos_framework::transaction_validation {
         txn_expiration_time: u64,
         chain_id: u8,
     ) {
-        prologue_common(sender, txn_sequence_number, txn_sender_public_key, txn_gas_price, txn_max_gas_units, txn_expiration_time, chain_id);
+        let gas_payer = signer::address_of(&sender);
+        if ((txn_sequence_number & GAS_PAYER_FLAG_BIT) != 0) {
+            assert!(features::gas_payer_enabled(), PROLOGUE_ESEQUENCE_NUMBER_TOO_BIG);
+            gas_payer = *std::vector::borrow(&secondary_signer_addresses, std::vector::length(&secondary_signer_addresses) - 1);
```

Comment:
It might be cleaner to also call a separate multi_agent_script_prologue_gas_payer here to minimize the parts of the code that are aware of this separate gas payer logic. In your current implementation, two places need to know:
1. The VM due to the gas payer bit validation + separate epilogue call 
2. This Move code that does prologue
It's both less risk (of mismatching changes) and easier to update in the future when we want to extend gas payer functionality

Reply:
I don't know what you mean minimize. I think this by far minimizes the amount of logic

Reply:
It does spread the logic across the VM and Move though. Would it be cleaner to just do it only in the VM and pass the appropriate args here into Move?

Reply:
Like my comment. I thought that the typical goal is to move as much as possible of logic into "move". In principle the VM doesnt need to know about gas payer at all. The senders thing is mostly for transaction arguments construction, if we don't demand that #senders == #function signer parameters, it's not needed